src/main/java/de/tum/cit/aet/artemis/communication/service/notifications/MailService.java

```diff
@@ -261,6 +263,18 @@ public void sendNotification(Notification notification, User user, Object notifi
             // posts use a different mechanism for the url
             context.setVariable(NOTIFICATION_URL, extractNotificationUrl(post, artemisServerUrl.toString()));
             subject = createAnnouncementText(notificationSubject, locale);
+
+            // Render markdown content of post to html
+            try {
+                Parser parser = Parser.builder().build();
+                HtmlRenderer renderer = HtmlRenderer.builder().build();
+                String postContent = post.getContent();
+                String renderedPostContent = renderer.render(parser.parse(postContent));
+                post.setContent(renderedPostContent);
```

Comment:
Are you sure there are no side effects when assigning the html content back to the post object?

Reply:
I did it this way because it is done like this when the notification is stored in `ConversationMessagingService:504`. I think some change may have caused the issue to arise, because the content is first set to the HTML, but then the notification Task only gets the serialized Post ID. This means the task has to re-query the Post content, returning it without the rendered HTML. The post is not saved after this in the code. To verify, I tried to check in the database after sending the notification and the original content stays unmodified.

Here the snipped from `ConversationMessagingService`:
<img width="794" alt="Bildschirmfoto 2024-10-19 um 10 27 57" src="https://github.com/user-attachments/assets/7118119f-3ad3-4f7f-8f42-f7cc54dd9353">

Reply:
However, if you think its too unsafe, i can create a new context variable and call that within the notification template.